src/MVVM/ViewModels/Navigation/ViewModelBase.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Threading.Tasks;
+
+using CodeMonkeys.Core.Interfaces.MVVM;
+using CodeMonkeys.Core.Interfaces.Navigation;
+using CodeMonkeys.Core.Interfaces.Navigation.ViewModels;
+using CodeMonkeys.MVVM.Factories;
+
+namespace CodeMonkeys.MVVM.ViewModels.Navigation
+{
+    public abstract class ViewModelBase :
```

Comment:
Why do we have 2 ViewModelBase classes? Does it make sense to name this one somehow different? For example: NavigationAwareViewModelBase. Dunno